packages/plugins/src/plugins/morpho/rewardsFetcher.ts

```diff
@@ -0,0 +1,115 @@
+    team has expressed intentions to open-source it in the future.
+*/
+
+function calculateRawRewardBalances(data: MorphoRewardsRes['data']) {
+  const balances: Record<Address, bigint> = {};
+
+  for (const entry of data) {
+    const address = entry.asset?.address
+      ? getAddress(formatEvmAddress(entry.asset?.address))
+      : null;
+    if (!address) continue;
+
+    let raw: bigint | null = null;
+
+    if (entry.type === 'uniform-reward' || entry.type === 'airdrop-reward') {
+      raw = BigInt(entry.amount?.claimable_now ?? '0');
+    } else if (entry.type === 'market-reward') {
+      raw = BigInt(entry.for_supply?.claimable_now ?? '0');
+    }
+
+    if (!raw || raw === BigInt(0)) continue;
+
+    balances[address] = (balances[address] || BigInt(0)) + raw;
+  }
+
+  return Object.entries(balances).map(([address, total]) => ({
+    address,
+    balance: total.toString(),
+  }));
+}
+
+export function getRewardsFetcher(networkId: EvmNetworkIdType): Fetcher {
+  const executor: FetcherExecutor = async (owner: string, cache: Cache) => {
+    const rewardsRes = await getRewards(owner, networkId);
+
+    const rawRewards = calculateRawRewardBalances(rewardsRes.data);
+
+    const tokenPricesMap = await cache.getTokenPricesAsMap(
+      rawRewards.map((rewardToken) => rewardToken.address),
+      networkId
+    );
+
+    /* 
+        Context:
+        The token price of Legacy Morpho is 0 (as indicated by the morpho team)
+        Typically the sonarwatch code will ignore positions without a token price.
+        It treats tokenPrice undefined and tokenPrice 0 the same
+        The code below works around this by assuming the token price of Legacy Morpho is 0
+     */
+    const rewardAssets = rawRewards.map((rewardAsset) => {
+      const tokenPrice = tokenPricesMap.get(rewardAsset.address);
+
+      // token price of Legacy Morpho is  0
+      const decimals = BigInt(tokenPrice?.decimals || 18);
+      // Note: used BigNumber here because BigInt exponential operations are not supported in the current TypeScript target version.
+      const balance = new BigNumber(rewardAsset.balance).dividedBy(
+        new BigNumber(10).pow(new BigNumber(decimals.toString()))
+      );
+
+      return tokenPriceToAssetToken(
+        rewardAsset.address,
+        balance.toNumber(),
+        networkId,
+        tokenPrice,
+        tokenPrice?.price || 0 // token price of Legacy Morpho should be 0
+      );
+    });
+
+    const rewardAssetsValue = getUsdValueSum(rewardAssets.map((a) => a.value));
+
+    const element: PortfolioElementMultiple = {
+      networkId,
+      platformId,
+      label: 'Rewards',
+      type: PortfolioElementType.multiple,
+      data: {
+        assets: rewardAssets,
+      },
+      value: rewardAssetsValue,
+    };
+
+    return [element];
+  };
+
+  return {
+    id: `${platformId}-${networkId}-rewards`,
+    networkId: NetworkId.ethereum,
```

Comment:
```suggestion
    networkId,
```

The above line has access to `networkId` variable; we should be able to use it too, right ?